docs/layers/identity/design-decisions/decide-on-aws-cli-login.mdx

```diff
@@ -0,0 +1,49 @@
+---
+title: "Decide on AWS CLI Login"
+sidebar_label: "AWS CLI Login"
+description: Decide on a CLI tool that enables AWS login and credentials via SAML IDP for CLI and web console access.
+---
+import Intro from '@site/src/components/Intro';
+import KeyPoints from '@site/src/components/KeyPoints';
+import Note from '@site/src/components/Note';
+import TaskList from '@site/src/components/TaskList';
+
+<Intro>
+Decide on a CLI tool that enables AWS login and credentials via SAML IDP for CLI and web console access.
+</Intro>
+
+## Problem
+
+Users need some way to login into AWS when using the CLI or applying Terraform changes. We have AWS Identity Center or AWS SAML setup for an AWS organization, but we need a way to login to AWS locally.
+
+There are a number of tools that can help with this, but we need to decide on one.
+
+### Option 1: Use the AWS CLI
+
+First of all, we could use the AWS CLI to login to AWS. This is the most basic way to login to AWS, but it requires a lot of manual steps and is not very user-friendly.
+
+### Option 2: Use Leapp
+
+Alternatively we could use Leapp by Noovolari. This is a tool that allows you to login to AWS using SAML and then automatically generates temporary credentials for you to use in the CLI. Once setup, Leapp makes it very easy to login to AWS and use the CLI, assume roles across your accounts with Role Chaining pre-configured, and even launch directly into the AWS web console.
+
+Leapp historically has been recommended by Cloud Posse and a popular choice for this use case, but recently Noovolari has announced that they are shutting down the service. The [leapp](https://github.com/Noovolari/leapp) project itself will continue to be supported, but the paid option will be shut down. This presents bad optics for the project and may make it less attractive to use for future customers.
+
+Plus Leapp requires several manual steps on initial setup, which has been a pain point for some users. See [How to Login to AWS (with Leapp)](/layers/identity/how-to-log-into-aws/) for more on the required setup and usage.
+
+Futhermore, Leapp requires setup steps outside of our Geodesic containers, which makes it less convenient to use and leads to user configuration errors.
+
+### Option 3: Use `aws-sso-cli`
+
+The most recent option we've come across is [aws-sso-cli](https://github.com/synfinatic/aws-sso-cli). This is a CLI tool that allows you to login to AWS using SAML and then automatically generates temporary credentials for you to use in the CLI. It is similar to Leapp, and is also open source and free to use. It also has a number of features that make it easier to use, such as the ability to login to multiple AWS accounts and roles at the same time.
```

Comment:
I don't believe it supports SAML, only AWS SSO :-/

Reply:
yeah this was before. We havent found a solution other than Leapp that accomplishes both. However most customers only use either/or, so we could recommend aws-sso-cli for AWS SSO users, and extend-roles or another for AWS SAML

Reply:
Let's add Option 4, which is saml2aws (SAML only)

And mention the caveats https://github.com/Versent/saml2aws?tab=readme-ov-file#caveats

It really only works well with Okta, which is pretty common. It works poorly with Google because of the "screen scraping" technique that triggers bot measures.

You can add 5 which is a browser plugin.